assets/js/settings/shared/get-setting.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Internal dependencies
+ */
+import { allSettings } from './settings-init';
+
+/**
+ * Retrieves a setting value from the setting state.
+ *
+ * @export
+ * @param {string} name                               The identifier for the
+ *                                                    setting.
+ * @param {mixed}  [fallback=false]                   The value to use as a
+ *                                                    fallback if the setting is
+ *                                                    not in the state.
+ * @param {function} [validation=( value ) => value]  A callback to use for
+ *                                                    validating or sanitizing
+ *                                                    the setting value.
+ * @returns {mixed}
+ */
+export function getSetting(
+	name,
+	fallback = false,
+	validation = ( value ) => value
```

Comment:
Hmm...

- Ya we should either provide `fallback` to the validation callback or run the callback on the fallback as well.  I think the former sounds better. Thoughts?
- I don't think it's necessary to split validation and sanitization callbacks.  The way I envisioned things working is that if something fails validation, then an error should be thrown (which means calling code would have to handle the error).  If we provide the fallback to the callback as a second arg, that also gives some more flexibility for how the consumer decides to handle failed validation (vs sanitization).  

One thing that could use improvement here is the name of the callback argument since it's not just intended for validation.  Any suggestions?

Reply:
- Is there an example of how an error would be raised from this, and caught?
- Not sure on naming. One benefit of supporting multiple callbacks is that you can be explicit in the naming i.e. santitize_callback, validate_callback, similar to how callbacks are added in the WP rest API. For a do-all callback, not really sure what you can call it...format? Just "callback"?
- Passing `fallback` is better IMO so you know when the fallback is used.

Reply:
- renamed callback to `filter` and adjusted docs to remove references of sanitization/validation.  Essentially its a flexible callback for consumers to use how they wish for filtering the value (typically could be used for sanitization).
- pass thru `fallback` to the `filter` callback.  This allows for consumers to potentially do validation as well and return the fallback value if validation fails.